Model/Sync/Catalog/Processor/Main.php

```diff
@@ -562,7 +562,7 @@ public function getExistingProductsFromAPI($url, $requestIds, $type)
         $response = $this->coreSync->sync('GET', $url, $data);
 
         $products = [];
-        if ($response && $response->getResponse()) {
+        if (is_object($response) && $response->getResponse()) {
```

Comment:
It is just for making sure $response is an object and it has the "status" property. We can remove is_object check from here, but it is something like a double safety to avoid any run time errors in case $response is null by chance.